embedded/store/verification.go

```diff
@@ -131,3 +131,59 @@ func leafFor(d [sha256.Size]byte) [sha256.Size]byte {
 	copy(b[1:], d[:])
 	return sha256.Sum256(b[:])
 }
+
+type EntrySpecDigest func(kv *EntrySpec) [sha256.Size]byte
+
+func EntrySpecDigestFor(version int) (EntrySpecDigest, error) {
+	switch version {
+	case 0:
+		return EntrySpecDigest_v1_1, nil
+	case 1:
+		return EntrySpecDigest_v1_2, nil
+	}
+
+	return nil, ErrIllegalArguments
+}
+
+func EntrySpecDigest_v1_1(kv *EntrySpec) [sha256.Size]byte {
+	b := make([]byte, len(kv.Key)+sha256.Size)
+
+	copy(b[:], kv.Key)
+
+	hvalue := sha256.Sum256(kv.Value)
+	copy(b[len(kv.Key):], hvalue[:])
+
+	return sha256.Sum256(b)
+}
+
+func EntrySpecDigest_v1_2(kv *EntrySpec) [sha256.Size]byte {
+	var mdbs []byte
+
+	if kv.Metadata != nil {
+		mdbs = kv.Metadata.Bytes()
+	}
+
+	mdLen := len(mdbs)
+	kLen := len(kv.Key)
+
+	b := make([]byte, sszSize+mdLen+sszSize+kLen+sha256.Size)
```

Comment:
For future consideration - sha256 has a Hasher interface t hat is io.Writer, we could do this like:
```golang
h := sha256.New()
binary.Write(h, binary.BigEndian, uint16(mdLen))
h.Write(mdbs)
....
return h.Sum(nil)
```

Reply:
yes, probably the gain in performance by allocating at once is not significative